ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -48,112 +30,149 @@ let update_mh_info interface =
+        StringSet.iter (fun addr -> start ~__context ~addr ()) to_start
+    | _, Local addresses ->
+        stop () ;
+        StringSet.iter (fun addr -> start ~__context ~addr ()) addresses
 
-let restart_stunnel ~__context ~accept =
-  info "Restarting stunnel (accepting connections on %s)" accept ;
-  (* cache `accept` so client can call `reconfigure_stunnel` easily *)
-  Mutex.execute stunnel_accept_m (fun () -> stunnel_accept := Some accept) ;
-  _restart_stunnel_no_cache ~__context ~accept
-
-let reconfigure_stunnel ~__context =
-  let f =
-    Mutex.execute stunnel_accept_m (fun () ->
-        match !stunnel_accept with
-        | None ->
-            fun () ->
-              D.warn
-                "reconfigure_stunnel: accept is not set, so not restarting \
-                 stunnel"
-        | Some accept ->
-            fun () -> _restart_stunnel_no_cache ~__context ~accept)
-  in
-  f ()
-
-let stop () =
-  debug "Shutting down the old management interface (if any)" ;
-  List.iter (fun i -> Http_svr.stop i) !management_interface_server ;
-  management_interface_server := [] ;
-  listening_all := false ;
-  listening_localhost := false ;
-  listening_himn := false
-
-(* Even though xapi listens on all IP addresses, there is still an interface appointed as
- * _the_ management interface. Slaves in a pool use the IP address of this interface to connect
- * the pool master. *)
-let start ~__context ?addr () =
-  let socket, accept =
-    match addr with
-    | None -> (
-        info "Starting new server (listening on all IP addresses)" ;
-        try
-          (* Is it IPv6 ? *)
-          let addr = Unix.inet6_addr_any in
-          (Xapi_http.bind (Unix.ADDR_INET (addr, Constants.http_port)), ":::443")
-        with _ ->
-          (* No. *)
-          let addr = Unix.inet_addr_any in
-          (Xapi_http.bind (Unix.ADDR_INET (addr, Constants.http_port)), "443")
-      )
-    | Some ip -> (
-        info "Starting new server (listening on %s)" ip ;
-        let addr = Unix.inet_addr_of_string ip in
-        let sockaddr = Unix.ADDR_INET (addr, Constants.http_port) in
-        ( Xapi_http.bind sockaddr
-        , match Unix.domain_of_sockaddr sockaddr with
-          | Unix.PF_INET6 ->
-              "::1:443"
-          | _ ->
-              "127.0.0.1:443" )
-      )
-  in
-  Http_svr.start Xapi_http.server socket ;
-  management_interface_server := socket :: !management_interface_server ;
-  restart_stunnel ~__context ~accept ;
-  if Pool_role.is_master () && addr = None then
-    (* NB if we synchronously bring up the management interface on a master with a blank
-       		   database this can fail... this is ok because the database will be synchronised later *)
-    Server_helpers.exec_with_new_task "refreshing consoles" (fun __context ->
-        Dbsync_master.set_master_ip ~__context ;
-        Dbsync_master.refresh_console_urls ~__context)
+  let current_mode = ref Off
```

Comment:
why not expose this externally using a function rather than a ref? like you did for `himn`